backend/server/data-cache.go

```diff
@@ -25,6 +28,8 @@ func newDataCache() *dataCache {
 }
 
 func (c *dataCache) Drop() {
+	c.Mutex.Lock()
```

Comment:
nit: maybe just c.Lock() and c.Unlock() as dataCache is lock.Mutex :)

Reply:
I'm fine either cause. I usually like to use the field names specifically to avoid ambiguity

Reply:
yup, it's just a style thingy :+1: